### DISCUSSION TITLE
Foreign key with ondelete set null

### DISCUSSION THREAD

**Discussion by jfernandezsierra** [fastapi/sqlmodel#613] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the SQLModel documentation, with the integrated search.
- [X] I already searched in Google "How to X in SQLModel" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to SQLModel but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to SQLModel but to [SQLAlchemy](https://github.com/sqlalchemy/sqlalchemy).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
class UUIDModel(SQLModel):
    uuid: uuid_pkg.UUID = Field(
        default_factory=uuid_pkg.uuid4,
        primary_key=True,
        index=True,
        nullable=False,
        sa_column_kwargs={
            "server_default": text("gen_random_uuid()"),
            "unique": True,
        },
    )

class Client(UUIDModel, SQLModel):
    number_login: str = Field(max_length=20, nullable=False, index=True)
    uuid_fiber_appointment: Optional[uuid_pkg.UUID] = Field(
        foreign_key="fiber_appointment.uuid",
        nullable=True,
    )
    fiber_appointment: "FiberAppointment" = Relationship(
        back_populates="clients",
        sa_relationship_kwargs={"lazy": "selectin"},
    )

class FiberAppointment(
    UUIDModel,
    SQLModel,
    table=True,
):
    __tablename__ = "fiber_appointment"
    appointment_datetime: datetime
    clients: List["Client"] = Relationship(
        back_populates="fiber_appointment",
    )
```

### Description

I need to delete a list of appointments and that the clients that had those appointments, in the foreign key field (uuid_fiber_appointment) be set to null

### Operating System

macOS

### SQLModel Version

0.0.8

### Python Version

Python 3.10.11

**Comment by FilipeMarch** [comment]
@jfernandez-majorel 

Have you tried adding `"cascade": ""` to `sa_relationship_kwargs`?

```python
class Client(UUIDModel, SQLModel):
    number_login: str = Field(max_length=20, nullable=False, index=True)
    uuid_fiber_appointment: Optional[uuid_pkg.UUID] = Field(
        foreign_key="fiber_appointment.uuid",
        nullable=True,
    )
    fiber_appointment: "FiberAppointment" = Relationship(
        back_populates="clients",
        sa_relationship_kwargs={"lazy": "selectin", "cascade": ""},
    )
```
Try it

**Comment by YuriiMotov** [comment, marked as answer]
Related section of docs: https://sqlmodel.tiangolo.com/tutorial/relationship-attributes/cascade-delete-relationships/#ondelete-with-set-null
